File: sdk/runanywhere-swift/Sources/RunAnywhere/Public/Errors/SDKError.swift
import Foundation

/// SDK-specific errors
public enum SDKError: LocalizedError {
    case notInitialized
    case notImplemented
    case modelNotFound(String)
    case loadingFailed(String)
    case generationFailed(String)
    case frameworkNotAvailable(LLMFramework)
    case downloadFailed(Error)
    case validationFailed(ValidationError)
    case routingFailed(String)

    public var errorDescription: String? {
        switch self {
        case .notInitialized:
            return "SDK not initialized. Call initialize(with:) first."
        case .notImplemented:
            return "This feature is not yet implemented."
        case .modelNotFound(let model):
            return "Model '\(model)' not found."
        case .loadingFailed(let reason):
            return "Failed to load model: \(reason)"
        case .generationFailed(let reason):
            return "Generation failed: \(reason)"
        case .frameworkNotAvailable(let framework):
            return "Framework \(framework.rawValue) not available"
        case .downloadFailed(let error):
            return "Download failed: \(error.localizedDescription)"
        case .validationFailed(let error):
            return "Validation failed: \(error.localizedDescription)"
        case .routingFailed(let reason):
            return "Routing failed: \(reason)"
        }
    }
}

File: sdk/runanywhere-android/src/main/java/com/runanywhere/sdk/errors/SDKError.kt
package com.runanywhere.sdk.errors

/**
 * SDK-specific errors
 */
sealed class SDKError : Exception() {
    object NotInitialized : SDKError() {
        override val message: String = "SDK not initialized. Call initialize(with:) first."
    }
    
    object NotImplemented : SDKError() {
        override val message: String = "This feature is not yet implemented."
    }
    
    data class ModelNotFound(val model: String) : SDKError() {
        override val message: String = "Model '$model' not found."
    }
    
    data class LoadingFailed(val reason: String) : SDKError() {
        override val message: String = "Failed to load model: $reason"
    }
    
    data class GenerationFailed(val reason: String) : SDKError() {
        override val message: String = "Generation failed: $reason"
    }
    
    data class FrameworkNotAvailable(val framework: String) : SDKError() {
        override val message: String = "Framework $framework not available"
    }
    
    data class DownloadFailed(val error: Throwable) : SDKError() {
        override val message: String = "Download failed: ${error.message}"
    }
    
    data class ValidationFailed(val error: ValidationError) : SDKError() {
        override val message: String = "Validation failed: ${error.message}"
    }
    
    data class RoutingFailed(val reason: String) : SDKError() {
        override val message: String = "Routing failed: $reason"
    }
}

/**
 * Validation error
 */
data class ValidationError(
    val field: String,
    val message: String,
    val code: String? = null
) 